core/katago/katago.go

```diff
@@ -48,49 +46,59 @@ type Analyzer struct {
 	bootWait sync.WaitGroup
 }
 
-var analyzerSingleton *Analyzer
-var once sync.Once
+// New creates a new analyzer. Before being used, the Start() must be called.
+func New(model string, configPath string, analysisThreads int) *Analyzer {
```

Comment:
This is much better in the non-singleton world; we dont need singletons so this is the right choice.

Reply:
Ack